frontend/src/components/ButtonComponent.vue

```diff
@@ -1,13 +1,39 @@
 <!-- Button to interact with -->
 <template>
-  <InteractableComponent @click="(e) => $emit('click', e)">
-    <!-- Slot for the text of the button -->
-    <slot></slot>
-  </InteractableComponent>
+  <div>
+    <div class="relative">
+      <div class="absolute w-full h-full text-center" v-if="loading">
+        <div
+          class="absolute h-full w-full bg-gray-300 opacity-70 rounded-md"
+        ></div>
```

Comment:
What is this empty div for?

Reply:
This is the 70-%-transparent background of the overlay behind the spinner. It is used to lift the spinner from the text inside the button.

Reply:
why isnt the spinner inside this div then? or the background property just on the spinner?

Reply:
Then the spinner would also be 70%-transparent

Reply:
also if the opacity was changed to the class `bg-opacity-70`?

Reply:
Maybe, I'll try

Reply:
Yes, that works, I'll change that